perception/autoware_image_projection_based_fusion/include/autoware/image_projection_based_fusion/fusion_node.hpp

```diff
@@ -72,81 +92,93 @@ class FusionNode : public rclcpp::Node
     const std::string & node_name, const rclcpp::NodeOptions & options, int queue_size);
 
 protected:
+  // Common process methods
   void cameraInfoCallback(
     const sensor_msgs::msg::CameraInfo::ConstSharedPtr input_camera_info_msg,
     const std::size_t camera_id);
-
-  virtual void preprocess(TargetMsg3D & output_msg);
-
-  // callback for Msg subscription
-  virtual void subCallback(const typename TargetMsg3D::ConstSharedPtr input_msg);
-
+  // callback for main subscription
+  void subCallback(const typename Msg3D::ConstSharedPtr input_msg);
   // callback for roi subscription
-
-  virtual void roiCallback(
-    const typename Msg2D::ConstSharedPtr input_roi_msg, const std::size_t roi_i);
-
-  virtual void fuseOnSingleImage(
-    const TargetMsg3D & input_msg, const std::size_t image_id, const Msg2D & input_roi_msg,
-    TargetMsg3D & output_msg) = 0;
-
-  // set args if you need
-  virtual void postprocess(TargetMsg3D & output_msg);
-
-  virtual void publish(const TargetMsg3D & output_msg);
-
+  void roiCallback(const typename Msg2D::ConstSharedPtr input_roi_msg, const std::size_t roi_i);
+  // callback for timer
   void timer_callback();
   void setPeriod(const int64_t new_period);
+  void exportProcess();
+
+  // 2d detection management methods
+  bool checkAllDet2dFused()
+  {
+    std::lock_guard<std::mutex> lock_det2d_flags(mutex_det2d_flags_);
+    for (const auto & det2d : det2d_list_) {
+      if (!det2d.is_fused) {
+        return false;
+      }
+    }
+    return true;
+  }
+  void setDet2dFused(Det2dManager<Msg2D> & det2d)
+  {
+    std::lock_guard<std::mutex> lock_det2d_flags(mutex_det2d_flags_);
+    det2d.is_fused = true;
+  }
+  void clearAllDet2dFlags()
+  {
+    std::lock_guard<std::mutex> lock_det2d_flags(mutex_det2d_flags_);
+    for (auto & det2d : det2d_list_) {
+      det2d.is_fused = false;
+    }
+  }
```

Comment:
We should move this functions to `private` instead of `protected` since it is not intended to be used by the inherited classes. (BTW I think it is applicable to other functions such as `cameraInfoCallback`, `subCallback`, etc... :melting_face: )

Reply:
reviewed member and method access range

fixed https://github.com/autowarefoundation/autoware.universe/pull/9789/commits/18bf3d062946aa3df91d9e6e86dfe9d7407abc79